tasks/main.yml

```diff
@@ -60,16 +60,29 @@
   become: true
   changed_when: false
 
+- name: Look for grub configuration files
+  stat:
+   path: "{{ item }}"
+  register: grub_search
+  become: true
+  loop: "{{ grub_configuration_files }}"
+
 - name: replace current grub config
   copy:
     backup: "{{ backup_grub_config_file }}"
     # /etc/grub2.cfg -> ../boot/grub2/grub.cfg
     follow: true
     content: "{{ grub_mkconfig_result.stdout }}"
-    dest: "{{ item }}"
+    dest: "{{ item.stat.path }}"
+  when: item.stat.exists
   become: true
   register: grub_update_result
-  with_items:
-    - /etc/grub2.cfg
-    - /etc/grub2-efi.cfg
+  loop: "{{ grub_search.results }}"
   notify: "{{ kernel_restart_handler }}"
+
+- name: Update /boot/loader/entries with grubby
+  # NOTE: This is the new fangled way: https://unix.stackexchange.com/questions/557841/what-is-the-correct-way-to-handle-default-kernelopts-in-fedora-31
+  # As grub reads configuration from /boot/loader/entries/.
+  command: grubby --update-kernel=ALL --args='{{ grub_cmdline_linux_new | join(" ") }}' --remove-args '{{ grub_cmdline_linux_remove | join(" ") }}'
```

Comment:
Why do we need to provide kernel args here and `/etc/default/grub`?

Reply:
The old method of updating grub.cfg no longer works but it does
configure the fallback value if no kernel command line options
are set in /boot/loader/entries. See this snippet extracted
from grub.cfg:

```
if [ -z "${kernelopts}" ]; then
  set kernelopts="root=LABEL=rootfs ro  console=tty0 console=ttyS0,115200 no_timer_check nofb nomodeset gfxpayload=text net.ifnames=1  "
fi
```


So I think that default value comes from `/etc/default/grub`

Reply:
Here is the comment for clarity:
```
# This section was generated by a script. Do not modify the generated file - all changes
# will be lost the next time file is regenerated. Instead edit the BootLoaderSpec files.
#
# The blscfg command parses the BootLoaderSpec files stored in /boot/loader/entries and
# populates the boot menu. Please refer to the Boot Loader Specification documentation
# for the files format: https://systemd.io/BOOT_LOADER_SPECIFICATION/.

# The kernelopts variable should be defined in the grubenv file. But to ensure that menu
# entries populated from BootLoaderSpec files that use this variable work correctly even
# without a grubenv file, define a fallback kernelopts variable if this has not been set.
#
# The kernelopts variable in the grubenv file can be modified using the grubby tool or by
# executing the grub2-mkconfig tool. For the latter, the values of the GRUB_CMDLINE_LINUX
# and GRUB_CMDLINE_LINUX_DEFAULT options from /etc/default/grub file are used to set both
# the kernelopts variable in the grubenv file and the fallback kernelopts variable.
if [ -z "${kernelopts}" ]; then
  set kernelopts="root=LABEL=rootfs ro  console=tty0 console=ttyS0,115200 no_timer_check nofb nomodeset gfxpayload=text net.ifnames=1 iommu=pt"
fi
```

Reply:
So I wasn't thinking it does no harm to set both. What do you reckon?

Reply:
I also found grubby (at least on RHEL 9) didn't set `/etc/default/grub`

Reply:
OK, I've stopped running grub-mkconfig on RHEL9 (kept old behaviour on <= RHEL 8). I've still added it to /etc/default/grub in case someone runs grub-mkconfig. What do you reckon?